### ISSUE TITLE
Progress indicator

### ISSUE THREAD

**Issue by igaray** [inaka/elvis#493]
When running elvis for the first time on a large project, it might take a while. It would be nice to have some (optional) progress indicator to know elvis hasn't left the building.

**Comment by igaray** [comment]
Likely requires implementing #382

**Comment by elbrujohalcon** [comment]
I think that a progress bar would be a shell-script thing, not a core thing. Although some adjustments to the core might be needed so it can trigger some events (or evaluate some callbacks) after each file is processed… so that tools like this CLI can use them to display a progress indicator.
If we implement that… we can also adjust `rebar3_lint` to use a similar approach.

**Comment by paulo-ferraz-oliveira** [comment]
Yes, I meant the core of it would have to come from `elvis_core`.
